public/javascripts/SVLabel/src/SVLabel/mission/MissionProgress.js

```diff
@@ -124,5 +88,47 @@ function MissionProgress (svl, gameEffectModel, missionModel, modalModel, neighb
         statusModel.setMissionCompletionRate(completionRate);
         statusModel.setProgressBar(completionRate);
         this._checkMissionComplete(currentMission, currentRegion);
+
+        // Survey prompt. Modal should display survey if
+        // 1. User has completed numMissionsBeforeSurvey number of missions
+        // 2. The user has just completed more than 30% of the current mission
+        // 3. The user has not been shown the survey before
+        if (completionRate > 0.30 && completionRate < 0.60) {
```

Comment:
What is the `completionRate < 0.6` check for?

Reply:
Fair question. 
Without this upper bound, the logic for the survey trigger is as follows: if the current mission complete rate is >0.3, and the user has completed exactly one mission, trigger the survey. This would trigger at the end of the 1st mission (because the completion rate of the current mission is >0.99) as soon as the db updates with this information. 0.6 is picked as a reasonable upper bound where I assume every mission yields some result between 0.3 and 0.6.

Reply:
I don’t see any issues with that. I can also play around some more to see if there is a “!ismissioncomplete” function I can leverage instead to make it the most accurate.